kernels/ihaskell/default.nix

```diff
@@ -1,27 +1,34 @@
 { writeScriptBin
 , haskellPackages
 , stdenv
+, customIHaskell ? null
 , extraIHaskellFlags ? ""
 , name ? "nixpkgs"
 , packages ? (_:[])
 }:
 
 let
-  ihaskellEnv = haskellPackages.ghcWithPackages (self: [ self.ihaskell ] ++ packages self);
+  # By default we use the ihaskell included in the `haskellPackages` set, but you can 
+  # also specify one explicitely in case the ihaskell you want to use resides somewhere
+  # else. Note that you will likely need an ihaskell which was built using the same
+  # ghc as the one used by `haskellPackages`.
+  ihaskell = if customIHaskell == null then haskellPackages.ihaskell else customIHaskell;
+
+  ghcEnv = haskellPackages.ghcWithPackages (self: packages self);
```

Comment:
The original code had the `ihaskell` package added to the GHC packages. Is there any reason for removing it?

Reply:
Ya, and I actually wanted to ask about that. Does ihaskell need to be importable in the notebooks? The reason I had taken that out is that haskell.nix has a notion of "components" which divide up a haskell project into separate derivations for the library and executable(s). This means that for calling ihaskell you want a different derivation than for passing the library to ghc, and passing in the customIHaskell in my changes causes haskell.nix's ghcWithPackages to fail.

Reply:
@guaraqe Actually, I just realized that I can just symlinkJoin them downstream and it works fine. I'll revert that line to also include ihaskell with the GHC packages by default again.